frontend/public/components/monitoring/alerting.tsx

```diff
@@ -599,10 +624,81 @@ export const AlertsDetailsPage = withFallback(
     // eslint-disable-next-line react-hooks/exhaustive-deps
     const labels: PrometheusLabels = React.useMemo(() => alert?.labels, [labelsMemoKey]);
 
+    const AlertStateDescription = ({ alertData }) => {
+      if (alertData && !_.isEmpty(alertData.silencedBy)) {
+        return (
+          <StateTimestamp
+            text={t('public~Ends')}
+            timestamp={_.max(_.map(alertData.silencedBy, 'endsAt'))}
+          />
+        );
+      }
+      if (alertState && alertState.activeAt) {
+        return <StateTimestamp text={t('public~Since')} timestamp={alertData.activeAt} />;
+      }
+      return null;
+    };
+
+    const getSourceKey = (source) => {
+      switch (source) {
+        case 'Platform':
+          return t('public~Platform');
+        case 'User':
+          return t('public~User');
+        default:
+          return source;
+      }
+    };
+
+    const SilenceTableRow: RowFunction<Silence> = ({ index, key, obj, style }) => {
```

Comment:
`SilenceTableRow` is another chunk of code that it getting duplicated with this change. I think we can leave it as a separate component that calls `useTranslation` itself.

Reply:
I tried this initially. If you use `useTranslation`, you get errors because you can only use `useTranslation` in a React function component. (This is a RowFunction typed component.) If you try to make it into a function component, the Table component that takes it as a Row prop complains that it's not a RowFunction. Hence the duplicates.

i18next.t seems to work ok, but I've gotten comments from Sam about it not being reactive before.